#! /usr/bin/env python
PACKAGE = "auv_codes2"  # Do not forget to change this line if required

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# gen.add("servoFrontRight",    int_t,    0, "Thruster_speed in microseconds", 1500,  1100, 1900)
# gen.add("servoFrontLeft",    int_t,    0, "Thruster_speed in microseconds", 1500,  1100, 1900)
# gen.add("servoMiddleRight",    int_t,    0, "Thruster_speed in microseconds", 1500, 1100, 1900)
# gen.add("servoMiddleLeft",    int_t,    0, "Thruster_speed in microseconds", 1500,  1100, 1900)
# gen.add("servoBackLeft",    int_t,    0, "Thruster_speed in microseconds", 1500,  1100, 1900)
# gen.add("servoBackRight",    int_t,    0, "Thruster_speed in microseconds", 1100,  1300, 1900)
# gen.add("mode_thrusters",   int_t,   0, "0-stop , 1-all 1600, 2- configuration mode, 3-camera mode, 4- imu sensor mode",  0,0,5)

gen.add("KP_yaw",    int_t,    0, "Value of KP for yaw", 1,  0, 1000)
gen.add("KI_yaw",    int_t,    0, "Value of KI for yaw", 1,  0, 1000)
gen.add("KD_yaw",    int_t,    0, "Value of KD for yaw", 1,  0, 1000)

gen.add("KP_depth",    int_t,    0, "Value of KP for depth", 1,  0, 1000)
gen.add("KI_depth",    int_t,    0, "Value of KI for depth", 1,  0, 1000)
gen.add("KD_depth",    int_t,    0, "Value of KD for depth", 1,  0, 1000)

exit(gen.generate(PACKAGE, "auv_codes2", "PID_yaw"))
